### PR TITLE
Supports array and function-type parameters

### LINKED ISSUE #48: Syntax error when mocking struct return type
```
typedef struct {...} struct_type;

struct_type some_function();

```
the empty mockup of some_function returns a 0, which does not work:
`return (struct_type)0;`

Instead, it needs to be the empty struct:
`return (struct_type){0};`

### PR DESCRIPTION
**Pull request by theHolgi**
Fixes #48

### CODE REVIEW

File: tests/data/mini_c_test/includes/a.h [hunk 1]
```diff
@@ -7,6 +7,11 @@ void a_step(void);
 extern int a_u1;
 extern int a_y1;
```
Comment (basejumpa) [inline tests/data/mini_c_test/includes/a.h:9]: `s_t` is introduced as type but is not used in any variable declaration. 

I suggest to add structs as simple variable, return type of a function and argument of a function, maybe also as a pointer (?) to the test data and add test(s) on it.

Comment (theHolgi) [reply tests/data/mini_c_test/includes/a.h:9]: Right, it was intended for a test [here](https://github.com/avengineers/hammocking/blob/d488a3d067134da08d871a219e885504faf4b14a/tests/hammocking_test.py#L256) but that didn't work.
We parse the variables by snippets, so the compiler does not see the include file when it encounters the variable
(that's why all types suddenly turned to `int`).
Should enhance the test here. It should more focus on the aspects of the template; the writer interfaces are better tested [here](https://github.com/avengineers/hammocking/blob/d488a3d067134da08d871a219e885504faf4b14a/tests/hammocking_test.py#L63).

Comment (basejumpa) [reply tests/data/mini_c_test/includes/a.h:9]: I see. That‘s reasonable.

From my side that‘s all good to go.

**Review by xxthunder — APPROVED** [review]
🚀

**Comment by xxthunder** [comment]
@theHolgi Very nice, thanks for solving this issue. Shall we merge?

**Pull request merged**